src/transports/local.c

```diff
@@ -512,8 +512,10 @@ static int foreach_reference_cb(git_reference *reference, void *payload)
 	git_revwalk *walk = (git_revwalk *)payload;
 	int error;
 
-	if (git_reference_type(reference) != GIT_REF_OID)
+	if (git_reference_type(reference) != GIT_REF_OID) {
+		git_reference_free(reference);
 		return 0;
+	}
```

Comment:
Would you care to put this into an extra PR? It's worthy to be included in v0.27.1, as I've just introduced that leak with v0.27.0